src/audio_core/algorithm/interpolate.cpp

```diff
@@ -30,40 +22,40 @@ std::vector<s16> Interpolate(InterpolationState& state, std::vector<s16> input,
         ratio = 1.0;
     }
 
-    if (ratio != state.current_ratio) {
-        const double cutoff_frequency = std::min(0.5 / ratio, 0.5 * ratio);
-        state.nyquist = CascadingFilter::LowPass(std::clamp(cutoff_frequency, 0.0, 0.4), 3);
-        state.current_ratio = ratio;
+    std::vector<s16> output;
+    output.resize(input.size() / ratio);
+
+    std::array<short, 512> lut;
```

Comment:
```suggestion
    const std::array<short, 512>& lut = [ratio] {
        const int step = static_cast<int>(ratio * 0x8000);
        if (step > 0xaaaa) {
            return curve_lut0;
        }
        if (step <= 0x8000) {
            return curve_lut1;
        }
        return curve_lut2;
    }();
```

This way the whole LUT isn't unnecessarily copied entirely each time the function is called.